### ISSUE TITLE
Add Dorico as a potential Music XML rader

### ISSUE THREAD

**Issue by EvilArchitech** [cuthbertLab/music21#1184]
**Motivation**

Especially with some of the larger educational software contract changes in the U.K. - as well as for all others - Dorico is one of the primary XML readers that users may have installed or prefer

**Feature summary**
Include Dorico as an optional XML reader during configuration.

**Proposed implementation**
Modify configure.py to include Dorico as an option exactly as it currently does for Sibelius, Finale,  and Musescore.

```
Optionally, describe the implementation or the interface, in code or pseudocode.
```
Actual code snippits:

# Declaration section existing
...
reMuseScoreApp = re.compile(r'MuseScore.*\.app', IGNORECASE)
reMuseScoreExe = re.compile(r'Musescore.*\\bin\\MuseScore.*\.exe', IGNORECASE)

# to be added
reDoricoApp = re.compile(r'Dorico.*\.app', IGNORECASE)
reDoricoExe = re.compile(r'Dorico.*\.exe', IGNORECASE)

# Change to __getMusicXMLReaderDarwin(self:
#Get all possible MusicXML Reader paths on Darwin (i.e., macOS)
        '''
        def comparisonFinale(x):
            return reFinaleApp.search(x) is not None

        def comparisonMuseScore(x):
            return reMuseScoreApp.search(x) is not None

        def comparisonFinaleReader(x):
            return reFinaleReaderApp.search(x) is not None

        def comparisonSibelius(x):
            return reSibeliusApp.search(x) is not None

        def comparisionDorico(x):
            return reDoricoApp.search(x) is not None    

        # order here results in ranks
        results = self._getDarwinApp(comparisonMuseScore)
        results += self._getDarwinApp(comparisonFinale)
        results += self._getDarwinApp(comparisonFinaleReader)
        results += self._getDarwinApp(comparisonSibelius)
        results += self._getDarwinApp(comparisonDorico)
        ...

# And similar for __GetMusicXMLReaderWin(self):
'''
        Get all possible MusicXML Reader paths on Windows
        '''
        def comparisonFinale(x):
            return reFinaleExe.search(x) is not None

        def comparisonMuseScore(x):
            return reMuseScoreExe.search(x) is not None and 'crash-reporter' not in x

        def comparisonSibelius(x):
            return reSibeliusExe.search(x) is not None

        def comparisonDorico(x):
            return reDoricoExe.search(x) is not None    

        # order here results in ranks
        results = self._getWinApp(comparisonMuseScore)
        results += self._getWinApp(comparisonFinale)
        results += self._getWinApp(comparisonSibelius)
        results += self._getWinApp(comparisonDorico)

**Intent**

[X ] I plan on implementing this myself.
[ ] I am willing to pay to have this feature added.
[ ] I am starting a discussion with the hope that community members will volunteer their time to create this. I understand that individuals work on features of interest to them and that this feature may never be implemented.

**Comment by jacobtylerwalls** [comment]
+1 from me. I look forward to a PR if you plan on sending one. Thanks! (Will you be able to test on a Windows system yourself?)

**Comment by mscuthbert** [comment]
Glad to have this in.  Which systems have you tested on?  It's rare that you can call an App file (which is a directory) with a filename argument.  But some came (Microsoft Word.app can be invoked with /safe I believe)

**Comment by EvilArchitech** [comment]
Good point - My Mac died a thermal death and I have only tested it on Windows.   Someone else can test for Mac on a Branch? I can submit a PR later today if that is acceptable.  Since it is just one file and its python, I can see if someone on the Dorico forum can help me out.

**Comment by EvilArchitech** [comment]
FWIW - prior to thermal death - I commonly invoke Dorico with an XML or MIDI file on Mac.

**Comment by benwiggy** [comment]
NB: there is a spelling error in the Mac function name 'comparisionDorico' should be 'comparisonDorico'.

**Comment by EvilArchitech** [comment]
Thanks Ben for testing and finding... Code is ready and on my fork/branch. PR submitted,
